pkgs/applications/emulators/duckstation/default.nix

```diff
@@ -45,13 +45,12 @@ stdenv.mkDerivation rec {
     SDL2
     curl
     libevdev
-    libpulseaudio
     libXrandr
     mesa
     qtbase
     vulkan-loader
     #wayland
-  ];
```

Comment:
Adding the backend libraries to `buildInputs` isn't necessary and does nothing. If you have Pipewire with JACK emulation enabled in your NixOS config, the module injects `$LD_LIBRARY_PATH` so things may appear to work but doesn't in other setups.

Reply:
Having the libraries in just `LD_LIBRARY_PATH` (but not in `buildInputs`) doesn't work, does that mean they have to be in both places?

Reply:
I just tried, and it's still using the vendored version of cubeb which is built statically so you need the backends in `buildInputs` to link and in `LD_LIBRARY_PATH` to lazy load (still, no connection to our `cubeb` package whatsoever). We will need to see if it's easy to coax it to use our cubeb.

Reply:
The point of this PR wasn't to vendor Cubeb (some of the emulators (like PCSX2) have also done modifications to it). I should have spcified it in the description...
You tested that the libraries don't indeed work if they are not in `LD_LIBRARY_PATH` (for DuckStation specifically)?

Reply:
Oh, I misread the intent of the PR. Then I think it's a bit misleading to use the `backendLibs` of the `cubeb` package when the emulator isn't actually using our `cubeb`.

> You tested that the libraries don't indeed work if they are not in LD_LIBRARY_PATH (for DuckStation specifically)?

Yes, they have to be in `LD_LIBRARY_PATH`. My environment has `LD_LIBRARY_PATH` globally (Pipewire's JACK emulation), and when launching with `LD_LIBRARY_PATH= ./result/bin/duckstation-qt` Duckstation fails to initialize audio.

Reply:
I tried `LD_LIBRARY_PATH= ./result/bin/duckstation-qt` (with the PR as it is), but the audio works on my end.
In fact, `echo $LD_LIBRARY_PATH` returns nothing.

Reply:
If I understand correctly, I concur with @zhaofengli. We shouldn't inject the `cubeb` packages for derivations that don't link on it (because e.g.: they use their own embedded copy). I still think the refactors of `cubeb` derivation makes sense though.

Reply:
Yeah, I'm not injecting the Cubeb package. I'm injecting the audio backends it uses.

Reply:
Yeah, I understand that. It is still kinda strange, because for a casual glance it looks like you're depending on the `cubeb` derivation, and also because it is not guarantee that every fork of cubeb out there supports the same backends.